backend/vxlan/vxlan_windows.go

```diff
@@ -11,14 +13,181 @@
+//  - N remote endpoints remote node (total endpoints =
 import (
+	"encoding/json"
+	"fmt"
+	"sync"
+
 	log "github.com/golang/glog"
+
+	"golang.org/x/net/context"
+
+	"errors"
+
+	"github.com/Microsoft/hcsshim"
+	"github.com/coreos/flannel/backend"
+	"github.com/coreos/flannel/pkg/ip"
+	"github.com/coreos/flannel/subnet"
 )
 
 func init() {
-	log.Infof("vxlan is not supported on this platform")
+	backend.Register("vxlan", New)
+}
+
+const (
+	minimumVNI = 4096
+	vxlanPort  = 4789
+)
+
+type VXLANBackend struct {
+	sm       subnet.Manager
+	extIface *backend.ExternalInterface
+	networks map[string]*network
+}
+
+func New(sm subnet.Manager, extIface *backend.ExternalInterface) (backend.Backend, error) {
+
+	be := &VXLANBackend{
+		sm:       sm,
+		extIface: extIface,
+		networks: make(map[string]*network),
+	}
+
+	return be, nil
+}
+
+func (be *VXLANBackend) RegisterNetwork(ctx context.Context, wg sync.WaitGroup, config *subnet.Config) (backend.Network, error) {
+	// TODO: are these used? how to pass to HNS?
+	cfg := struct {
+		name          string
+		macPrefix     string
+		VNI           int
+		Port          int
+		GBP           bool
+		DirectRouting bool
+	}{
+		name:      "vxlan0",
+		VNI:       minimumVNI,
+		Port:      vxlanPort,
+		macPrefix: "0E-2A",
+	}
+
+	if len(config.Backend) > 0 {
+		if err := json.Unmarshal(config.Backend, &cfg); err != nil {
+			return nil, fmt.Errorf("error decoding VXLAN backend config: %v", err)
+		}
+	}
+
+	if cfg.VNI < minimumVNI {
+		return nil, fmt.Errorf("invalid VXLAN backend config. VNI [%v] must be greater than or equal to %v on Windows", cfg.VNI, minimumVNI)
+	}
+
+	if cfg.Port != vxlanPort {
+		return nil, fmt.Errorf("invalid VXLAN backend config. Port [%v] is not supported on Windows. Omit the setting to default to port %v", cfg.Port, vxlanPort)
+	}
+
+	if cfg.DirectRouting == true {
+		return nil, errors.New("invalid VXLAN backend config. DirectRouting is not supported on Windows")
+	}
+
+	if cfg.GBP == true {
+		return nil, errors.New("invalid VXLAN backend config. GBP is not supported on Windows")
+	}
+
+	if cfg.macPrefix == "" || len(cfg.macPrefix) != 5 || cfg.macPrefix[2] != '-' {
+		return nil, fmt.Errorf("invalid VXLAN backend config. macPrefix [%v] is invalid, prefix must be of the format xx-xx e.g. 0E-2A", cfg.macPrefix)
+	}
+
+	log.Infof("VXLAN config: %+v", cfg)
+
+	n := &network{
+		extIface:  be.extIface,
+		sm:        be.sm,
+		name:      be.extIface.Iface.Name,
+		macPrefix: cfg.macPrefix,
+	}
+
+	attrs := subnet.LeaseAttrs{
+		PublicIP:    ip.FromIP(be.extIface.ExtAddr),
+		BackendType: "vxlan",
+	}
+
+	l, err := be.sm.AcquireLease(ctx, &attrs)
+	switch err {
+	case nil:
+		n.lease = l
+
+	case context.Canceled, context.DeadlineExceeded:
+		return nil, err
+
+	default:
+		return nil, fmt.Errorf("failed to acquire lease: %v", err)
+	}
+
+	// check if the network exists and has the expected settings?
+	networkName := cfg.name
+	createNetwork := true
+	addressPrefix := config.Network
```

Comment:
is this not supposed to be
```
	addressPrefix := l.Subnet
	networkGatewayAddress := l.Subnet.IP + 1
```
lease information is acquired but not used during HNSNetwork creation afaik I can understand?

Reply:
@alekssaul  Yes, you are correct